Add tutorial for Cache filter for Python

EsipovPA opened the pull request:
## Description

Added a tutorial for Cache message filter class for Python programming launguage.

<!--
Please include a summary of the changes and the related issue.
Highlight any key points or areas of concern.
-->

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.*
-->

Partially fixes #130 

### Did you use Generative AI?

### Additional Information

EsipovPA committed:
Add tutorial for Cache filter for Python

Signed-off-by: EsipovPA <esipov.p@mail.ru>

EsipovPA commented:
Noticed the formatting issues in python code snippets. I've copied this examples from VSCode to Formiko editor part by part. In the editor it looked Ok, but turns out, it is not. Will fix asap.

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,309 @@
+Overview
+~~~~~~~~
+
+This tutorial demonstrates how to use the ``message_filters.Cache`` class in ROS 2 using Python. The ``Cache`` filter stores a time history of messages and allows querying based on timestamps.
```
One sentence per line
```suggestion
This tutorial demonstrates how to use the ``message_filters.Cache`` class in ROS 2 using Python. 
The ``Cache`` filter stores a time history of messages and allows querying based on timestamps.
```

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,309 @@
+~~~~~~~~~~~~~
+This tutorial assumes you have a working knowledge of ROS 2
+
+If you have not done so already `create a workspace <https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Creating-A-Workspace/Creating-A-Workspace.html>`_ and `create a package <https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Creating-Your-First-ROS2-Package.html>`_
+
+
+1. Create an Example Node with Includes
```
```suggestion
If you have not done so already `create a workspace <https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Creating-A-Workspace/Creating-A-Workspace.html>`_ and `create a package <https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Creating-Your-First-ROS2-Package.html>`_

1. Create an Example Node with Includes
```

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,309 @@
+
+1. Create an Example Node with Includes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Let's assume, you've already created an empty ros package for Python. The next step is to create a new Python file inside your package, e.g., ``cache_tutorial.py``, and create an example node:
```
```suggestion
Let's assume, you've already created an empty ros package for Python. 
The next step is to create a new Python file inside your package, e.g., ``cache_tutorial.py``, and create an example node:
```

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,309 @@
+            qos = QoSProfile(depth=10)
+
+            # Publisher and Subscriber
+		        self.publisher = self.create_publisher(String, 'input', qos_profile)
```
identation looks wrong

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,309 @@
+
+.. code-block:: python
+
+		        self.subscriber_filter = Subscriber(
```
identation

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,309 @@
+            qos = QoSProfile(depth=10)
+
+            # Publisher and Subscriber
+		        self.publisher = self.create_publisher(String, 'input', qos_profile)
```
indentation

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,309 @@
+.. code-block:: python
+
+            # Simulate publishing via timer
+		        self.publisher_timer = self.create_timer(
```
identation

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,309 @@
+
+.. code-block:: python
+
+		    def publisher_timer_callback(self):
```
identation

ahcorde reviewed (changes requested):
Add also a link in `tutorials.rst`

EsipovPA commented:
@ahcorde Thank you for providing a review so quickly! I've updated the code, fixed the indentations and added the other changes, you've recommended. It would be great if you could have another look at this PR at your convenience.

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,306 @@
+            self.cahce_filter = Cache(
+                self.subscriber_filter,
+                cache_size=5,
+                allow_headerless=True,  # To allow cahcing basic String message
```
```suggestion
                allow_headerless=True,  # To allow caching basic String message
```

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,306 @@
+            self.cahce_filter = Cache(
+                self.subscriber_filter,
+                cache_size=5,
+                allow_headerless=True,  # To allow cahcing basic String message
```
```suggestion
                allow_headerless=True,  # To allow caching basic String message
```

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,306 @@
+                "/example/topic",
+                qos_profile=qos_profile,
+            )
+            self.cahce_filter = Cache(
```
```suggestion
            self.cache = Cache(
```

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,306 @@
+
+	[INFO] [1750884527.235426721] [cache_node]: Cache filters cache is empty
+	
+As there were no messages published yet, and the cahce is empty.
```
```suggestion
As there were no messages published yet, and the cache is empty.
```

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,306 @@
+
+.. code-block:: bash
+
+    colcon build && . install/setup.zsh
```
```suggestion
.. tabs::

  .. group-tab:: Linux

    .. code-block:: console
        $ colcon build && . install/setup.bash

  .. group-tab:: macOS

    .. code-block:: console
        $ colcon build && . install/setup.bash

  .. group-tab:: Windows

    .. code-block:: console
        $ colcon build
        $ call C:\dev\ros2\local_setup.bat
```

EsipovPA commented:
@ahcorde Thanks again for updates! I've fixed the typos and changed the naming as you've suggested, and updated the `build/install` section.

ahcorde commented on doc/Tutorials/Cache-Python.rst:
```diff
@@ -0,0 +1,317 @@
+
+From the root of your workspace:
+
+.. tab:
```
```suggestion
.. tabs:
```

EsipovPA replied:
Yes, I've noticed this right after pushing. Already fixed. :)

ahcorde committed:
Update doc/Tutorials/Cache-Python.rst

Signed-off-by: Alejandro Hernández Cordero <ahcorde@gmail.com>

ahcorde commented:
Thank you / Спасибо  @EsipovPA

ahcorde commented:
https://github.com/Mergifyio backport kilted jazzy humble
